src/Native/System.Native/pal_io.cpp

```diff
@@ -149,7 +149,8 @@ static void ConvertFileStatus(const struct stat_& src, FileStatus* dst)
 extern "C" int32_t Stat(const char* path, FileStatus* output)
 {
     struct stat_ result;
-    int ret = stat_(path, &result);
+    int ret;
+    while (CheckIo(ret = stat_(path, &result)));
```

Comment:
And https://www.python.org/dev/peps/pep-0475/ lists fstat as a system function that would need modification for handling EINTR.